File: edu-ios/Sources/Networking/AppConfig.swift
import Foundation

// Single source of truth for the backend base URL, shared by APIClient (which
// appends /api/mobile/v1) and by views that load site assets like category
// images from the site root (/categories/<slug>.png).
enum AppConfig {
    static let baseURL: String = {
        if let env = ProcessInfo.processInfo.environment["API_BASE_URL"], !env.isEmpty {
            return env
        }
        // Key off the environment, not the build config: the Simulator can reach the
        // Mac's dev server at localhost, but a physical device cannot — it needs the
        // deployed production API (works over any network, no LAN IP needed).
        #if targetEnvironment(simulator)
        return "http://localhost:3000" // Simulator -> Mac's `npm run dev`
        #else
        return "https://timpson-lyceum.vercel.app" // Physical device (+ Release) -> production
        #endif
    }()

    /// URL for a static asset served from the site root, e.g. "/categories/history.png".
    static func assetURL(_ path: String) -> URL? {
        URL(string: baseURL + path)
    }
}

File: edu-android/app/src/main/java/com/timpsonlyceum/lyceum/net/AppConfig.kt
package com.timpsonlyceum.lyceum.net

/**
 * Single source of truth for the backend base URL, shared by [ApiClient] (which
 * appends /api/mobile/v1) and by screens that load site assets from the site
 * root (/categories/<slug>.png).
 *
 * The iOS app keys off the simulator and points at the Mac's dev server; here we
 * default to production instead, because it works from an emulator, a physical
 * device, and CI alike with no LAN address to keep in sync. Point [override] at
 * http://10.0.2.2:3000 to reach `npm run dev` on the host from an emulator.
 */
object AppConfig {
    var override: String? = null

    val baseUrl: String
        get() = override ?: "https://timpson-lyceum.vercel.app"

    /** URL for a static asset served from the site root, e.g. "/categories/history.png". */
    fun assetUrl(path: String): String = baseUrl + path
}
